xml/art_virtualization-best-practices.xml

```diff
@@ -3241,6 +3241,162 @@ LC_ALL=C PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
       </para>
 <screen>&prompt.root;<command>echo 1 > /sys/devices/system/cpu/cpu<replaceable>X</replaceable>/online</command></screen>
     </sect2>
+
+    <sect2 xml:id="sec-vm-guest-scsi-pr">
+      <title>SCSI3 persistent reservation (S3-PR) on a multipathed device</title>
+      <para>
+        SCSI persistent reservations allow restricting access to block devices
+        to specific initiators in a shared storage setup.  When implementing
+        clustering of &vmguest;, it is a common requirement for &vmguest; to
+        send persistent reservation SCSI commands.  However, the operating
+        system restricts sending these commands to unprivileged programs
+        because incorrect usage can disrupt regular operation of the storage
+        fabric. QEMU's SCSI passthrough devices <literal>scsi-block</literal>
+        and <literal>scsi-generic</literal> support passing guest persistent
+        reservation requests to a privileged external helper program.
+        The <literal>qemu-pr-helper</literal> is that external helper, It needs
+        to start before QEMU, creates a listener socket which will accept
+        incoming connections for communication with QEMU.
+      </para>
+      <para>
+        Proper support of persistent reservation for multipath devices requires
+        communication with the multipath daemon, so that the reservation is
+        registered and applied when a path comes up.
+        You should start the multipathd service on the host.
+<screen>&prompt.user;&sudo; systemctl start multipathd.service</screen>
+      </para>
+      <note>
+        <title>Live migration scenario with multipathed devices</title>
+        <para>
+          We recommend using the multipath alias instead of wwid, It's helpful
+          in &vmguest; live migration scenario, It can make sure the storage path
+          are identical between source host and destination host.
+        </para>
+        <para>
+          About aliases for multipath, Please refer to
+          <link
+          xlink:href="https://documentation.suse.com/sles/15-SP5/html/SLES-all/cha-multipath.html#sec-multipath-alias"/>
+        </para>
+      </note>
+      <para>
+        In the following example we will share how to use SCSI3 persistent
+        reservation on a multipathed device:
+      </para>
+      <para>
+        In host:
+      </para>
+      <para>
+        Assume that we have such a multipathed disk:
+      </para>
+<screen>&prompt.user;&sudo; multipath -ll
+storage1 (36589cfc000000537c47ad3eb2b20216e) dm-6 TrueNAS,iSCSI Disk
+size=50G features='0' hwhandler='1 alua' wp=rw
+|-+- policy='service-time 0' prio=50 status=active
+| `- 16:0:0:0 sdg 8:96  active ready running
+`-+- policy='service-time 0' prio=50 status=enabled
+  `- 17:0:0:0 sdh 8:112 active ready running</screen>
+      <para>
+        Add the disk element in the &vmguest; configuration file (by running
```

Comment:
this is probably another step in the step-by-step "how to add scsi persistent reservation on multipathed i/o". Then the example section can be separate?

Reply:
Sorry I didn't get your point. From libvirt's angle, I already presented a disk element example that contains everything needed for correclty configure disk element for scsi persistent reservation. Could you please show me a detailed example about any missing part you mentioned?

Reply:
it's about structuring it a bit:

1) What is Storage multipathed I/O ? -> Storage doc reference

2) What are SCSI persistent reservations, and why are they important for virtualization?

3) How do I set up storage multipath I/O SCSI persistent reservations for my VMs?

Step-by-Step guide:

1 - install this, this, that additional package required for SCSI persistent reservations (if needed)
2 - configure your VM adding this to the domain specification (see libvirt domain xml documentation at ...)
3 - start this service or assure that it is running: systemctl ...
4 - do this other step ...
5 - use this command to check that everything is set up ok: ...

DONE!

Note: be careful as you do XYZ, you want to avoid this pitfall: ....

4) Examples:

And here you put the practical examples you already provided, for additional clarifications.

Reply:
I'll rewrite this doc, then send a new pr, to meet the suggestions in all of comments as much as I can.

Or:
I already present the material for writing, How about you write this doc, submit a PR, then Jim and I review?
IMO this way can speed up the whole process.

Reply:
Hi Lin, maybe if you want to send a new version I can take a look at that and if still I think there should be more structure to it we can invert our roles as you suggest? Thanks

Reply:
No problem, I'll rewrite it then send a new one.
Thanks for your review and comments.